### PR TITLE
[1LP][RFR] Add retries to certificate copy while copy fails network issue occurs

### PR DESCRIPTION
**Pull request by shalomnaim1**
{{ pytest: cfme/tests/containers/test_ssl_providers.py::test_add_mertics_provider_ssl --use-provider ocp-36-hawk }}
skipping add metrics endpoint test for providers without metrics endpoint

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 717c7b3cae7e0f0fa4a912f1c347e71ff203e984:

`cfme/tests/containers/test_ssl_providers.py`:
- :heavy_exclamation_mark: Line 57:30: **W601** *.has_key() is deprecated, use 'in'*
- :bangbang: Line 124:101: **E501** *line too long (103 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f19a27e766f886df43859cf717571b488fe1df2c:

`cfme/containers/provider/openshift.py`:
- :bangbang: Line 337:101: **E501** *line too long (102 > 100 characters)*
- :bangbang: Line 363:101: **E501** *line too long (106 > 100 characters)*

`cfme/tests/containers/test_ssl_providers.py`:
- :bangbang: Line 57:8: **E713** *test for membership should be 'not in'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c038cb3cdd7348ab1f9172dc4690536cfd7223cf:

`cfme/containers/provider/openshift.py`:
- :red_circle: Line 16:1: **F401** *'urllib2.URLError' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/containers/provider/openshift.py [hunk 1]
```diff
@@ -328,6 +328,20 @@ def sync_ssl_certificate(self):
              None
         """
 
+        def copy_certificate():
```
Comment (rlbabyuk) [inline cfme/containers/provider/openshift.py]: Required: def _copy_certificate():

Comment (shalomnaim1) [reply cfme/containers/provider/openshift.py]: @rlbabyuk Done

**Comment by shalomnaim1** [comment]
@rlbabyuk Could you please review the fixes I did?

File: cfme/containers/provider/openshift.py [hunk 2]
```diff
@@ -328,6 +328,20 @@ def sync_ssl_certificate(self):
...
 
+        def _copy_certificate():
+            is_succeed = True
+            try:
+                # Copy certificate to the appliance
+                provider_ssh.get_file("/etc/origin/master/ca.crt", "/tmp/ca.crt")
+                appliance_ssh.put_file("/tmp/ca.crt",
+                                       "/etc/pki/ca-trust/source/anchors/{crt}".format(
+                                           crt=cert_name))
+            except Exception:
```
Comment (rlbabyuk) [inline cfme/containers/provider/openshift.py]: Required: Please specify more exact type of Exception

Comment (shalomnaim1) [reply cfme/containers/provider/openshift.py]: @rlbabyuk  Well you right, the exception that I need to catch here is ```URLError```,

I found that this exception is in different packages in Python 2.7 and 3.6.

In Python 3.6 you can find this exception in ```urllib.error``` and in Python 2.7 it exists in ```urllib2```

Do you know how can I manage this situation?

Comment (shalomnaim1) [reply cfme/containers/provider/openshift.py]: @rlbabyuk Do you think that is a good solution?

```python
try:
    from urllib2 import URLError
except ImportError as ex:
    from urllib.error import URLError
```

Comment (rlbabyuk) [reply cfme/containers/provider/openshift.py]: please use import from six.moves.urllib.error import URLError

Comment (shalomnaim1) [reply cfme/containers/provider/openshift.py]: @rlbabyuk Done

**Comment by shalomnaim1** [comment]
@rlbabyuk Please approve

File: cfme/containers/provider/openshift.py [hunk 3]
```diff
@@ -14,6 +14,7 @@
 from cfme.utils.wait import wait_for, TimedOutError
 from . import ContainersProvider, ContainersProviderDefaultEndpoint, ContainersProviderEndpointsForm
 
+from six.moves.urllib.error import URLError
```
Comment (rlbabyuk) [inline cfme/containers/provider/openshift.py]: Required: this import should be moved to the top(to line 4)

**Pull request merged**